packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -135,8 +142,15 @@ export const GridColumnHeaderItem = ({
   );
   const columnMenuIconButton = <ColumnHeaderMenuIcon column={column} />;
 
+  React.useLayoutEffect(() => {
+    if (hasFocus && apiRef!.current.getState().columnMenu) {
+      headerCellRef.current!.focus();
+    }
+  });
```

Comment:
I don't get why we force the focus at each render. This seems to be the same problem as with the cell. I would be surprised if this is ever correct. For instance, could it be why the column menu isn't usable with the keyboard?

I even wonder if this focus state should ever be in the state. The focus is typically a state you might always prefer to manage imperatively.

Reply:
Am not at each render. This is to check which cell is currently the one that should be focused on and not steal the focus when the column menu is opened. The cell that is to be focused is stored in the state.